tests/stream_commands_test.py

```diff
@@ -0,0 +1,574 @@
+from collections import OrderedDict
+
+from time import sleep
+
+import pytest
+import asyncio
+
+from aioredis import ReplyError
+
+
+@asyncio.coroutine
+async def add_message_with_sleep(redis, loop, stream, fields):
+    await asyncio.sleep(0.2, loop=loop)
+    result = await redis.xadd(stream, fields)
+    return result
+
+
+@pytest.mark.run_loop
+@pytest.redis_version(999, 999, 999, reason="Streams only available on redis "
+                                            "unstable branch")
+async def test_xadd(redis, server_bin):
+    fields = OrderedDict((
+        (b'field1', b'value1'),
+        (b'field2', b'value2'),
+    ))
+    message_id = await redis.xadd('test_stream', fields)
+
+    # Check the result is in the expected format (i.e: 1507400517949-0)
+    assert b'-' in message_id
+    timestamp, sequence = message_id.split(b'-')
+    assert timestamp.isdigit()
+    assert sequence.isdigit()
+
+    # Read it back
+    messages = await redis.xrange('test_stream')
+    assert len(messages) == 1
+    message = messages[0]
+    assert message[0] == message_id
+    assert message[1] == OrderedDict([
+        (b'field1', b'value1'),
+        (b'field2', b'value2')]
+    )
+
+
+@pytest.mark.run_loop
+@pytest.redis_version(999, 999, 999, reason="Streams only available on redis "
+                                            "unstable branch")
+async def test_xadd_maxlen_exact(redis, server_bin):
+    message_id1 = await redis.xadd('test_stream', {'f1': 'v1'})  # noqa
+    sleep(0.001)  # Ensure the millisecond-based message ID increments
```

Comment:
Do you really need a thread-blocking sleep?

Reply:
Ha, I certainly don't :-)